### ISSUE TITLE
Test dataloader

### ISSUE THREAD

**Issue by Y-Claw** [xiaobai1217/RepetitionCounting#13]
I noticed there is no test dataloader in your training script. Could you give me a dataloader include test. Because I think test dataloader should be a little different when you sample videos.

**Comment by xiaobai1217** [comment]
Hi, 

In run_demo.py, we show how to evaluate on one single video. You can simply modify it to test on all videos.

**Comment by Y-Claw** [comment]
Thank you very much. By the way, may I ask if visual checkpoint you provide in github can achieve the mae and OBO values in the paper with fixed sampling rate? Or I should use your code choose sample rate first?

**Comment by xiaobai1217** [comment]
Yes I remember the checkpoint I released can achieve the mae and OBO for fixed sampling rate. Since many Youtube videos are not available anymore, you may have different test sets as me. So the performances may be slightly different. But the trend should be the same.

**Comment by Y-Claw** [comment]
With visual checkpoint you provide, I've tried use get_visual_count() from util with fixed sampling rate(1,2) and round sampling rate (round(frame_num / 64)), and tried generate output just like code in train.py, but all get mae around 0.5. For the missing video problem, I miss about 40 data, which is less than 10% of all test data, so I think larger mae is not all due to missing data. Could you provide a simple script of evaluating visual model that approximates the results of the paper. It will be very helpful, and I'll really really appreciate it.

**Comment by xiaobai1217** [comment]
If you want to get the performance reported in Table 7, you should not refer to the code in train.py. You can simply preserve the code related to the visual part in run_demo.py and set "outputs = get_visual_count(video1, 1, model, tensor)", i.e. sample rate = 1 for example.

**Comment by xiaobai1217** [comment]
import torch
import numpy as np
from sync_batchnorm import convert_model
import models
import VGGSound
from VGGSound.test import args_dict
from VGGSound.model import AVENet
import torch.nn as nn
import librosa
from scipy import signal
from util import get_sr, get_audio_count, get_conf, get_visual_count, read_video


np.random.seed(0)
torch.manual_seed(0)
torch.backends.cudnn.enabled = False  # 0.811

device = torch.device("cuda" if torch.cuda.is_available() else "cpu")

#-----------------------------------------------load models trained on Countix-AV-------------------------------------------
model = models.video.r2plus1d_18(pretrained=True)
model.fc = torch.nn.Linear(512,34*41)
model.fc2 = torch.nn.Linear(512,41)
model = convert_model(model)


if device.type == "cuda":
    model = torch.nn.DataParallel(model)

model = model.cuda()
checkpoint = torch.load("visual_checkpoint.pt")
model.load_state_dict(checkpoint['state_dict'])
model.eval()

tensor = torch.Tensor(np.arange(2,36)).type(torch.FloatTensor).cuda().unsqueeze(0)

video1 = "eEG7ZOQG6LM.00.mp4"
video1,fps = read_video(video1)
video1 = video1/255.0
video1 = (video1 - np.array([0.485,0.456,0.406]).reshape((1,1,1,3)))/np.array([0.229,0.224,0.225]).reshape((1,1,1,3))
#when we generated the video segment, we preserved some background context, so we need to do this line to remove the background
video1 = video1[10:265]

outputs = get_visual_count(video1, 1, model, tensor) # set sample rate to 1

groundtruth=8
mae_err = abs(outputs-groundtruth) / groundtruth
if abs(outputs-groundtruth) <= 1:
    OBO = 1
else:
    OBO = 0

print(mae_err, OBO)

**Comment by xiaobai1217** [comment]
This should be the code for testing what you want

**Comment by Y-Claw** [comment]
Thank you very much for answering my questions. I tried script with load checkpoint or train a model by myself. But it's still just get mae 0.4 - 0.5. I'll have more test. Finally, I want to make sure that you only selected the corresponding video clips according to Countix during the video pre-processing and used 112*112 images during the training.

**Comment by xiaobai1217** [comment]
Yes you are right. With fixed sampling rates, it is correct that they have higher mae, so I think 0.4-0.5 is the normal range.

**Comment by Y-Claw** [comment]
You mean if i want to reach 0.33 mae that report in paper, I should try your method of choosing sample rate.

**Comment by xiaobai1217** [comment]
yes!

**Comment by Y-Claw** [comment]
Thank you. I'll try it these day.

**Comment by DWhettam** [comment]
@Y-Claw were you able to reproduce the results in the paper with this method? Thanks!

**Comment by Y-Claw** [comment]
@DWhettam It is possible to achieve the results in the paper by using the checkpoint provided by them, but I did not succeed in retraining myself. It's been a while since I've been looking at counting, and I haven't been trying to reproduce.

**Comment by xiaobai1217** [comment]
Hi,

I created a google drive to save the videos of countix I have now. The number should be much higher than what can be downloaded now. Hope this can help. 
https://drive.google.com/file/d/1pp-Akj0Ydyh-AIElEbIx8Bp-f-DqQG3b/view?usp=share_link

**Comment by DWhettam** [comment]
@Y-Claw Do you have your code for testing? I am struggling to reproduce using the provided checkpoints
